### PR TITLE
Example-app-base: TEST - DO NOT MERGE

### PR DESCRIPTION
**Pull request by Vitalius1**
Testing a query parameter conditional rendering for docs.microsoft. Locally works but now need a dev deploy link to test it.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9333)

### CODE REVIEW

**Comment by Vitalius1** [comment]
@ecraig12345 can not make the build pass because of the ssr-tests failing on me when using the `location` variable. Any ideas how to bypass this? Isn't the `Router` also using `location` and passing the tests? What I am missing here? Thanks.

File: packages/example-app-base/src/components/ComponentPage/ComponentPage.tsx [hunk 1]
```diff
@@ -49,21 +49,23 @@ export class ComponentPageBase extends React.PureComponent<IComponentPageProps>
...
     return (
       <div className={css(classNames.root, className)}>
         <div className={componentName}>
-          {this._getPageHeader()}
+          {!showOnlyExamples && this._getPageHeader()}
           <div className={classNames.body}>
-            {this._getComponentStatusBadges()}
-            {this._getOverview()}
-            {this._getBestPractices()}
+            {!showOnlyExamples && this._getComponentStatusBadges()}
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ComponentPage/ComponentPage.tsx]: Is there any way you can avoid separately checking `!showOnlyExamples` on every line? Probably not a perf issue or anything, more just readability (arguably).

File: packages/example-app-base/src/components/ComponentPage/ComponentPage.tsx [hunk 2]
```diff
@@ -49,21 +49,23 @@ export class ComponentPageBase extends React.PureComponent<IComponentPageProps>
   public render() {
     const { componentName, className, otherSections, styles, theme } = this.props;
 
+    const showOnlyExamples = location && location.hash.indexOf('docsExample=true') > -1;
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ComponentPage/ComponentPage.tsx]: I'm not sure this makes sense--how would you put a query parameter into the hash (along with the route)?

File: packages/example-app-base/src/components/App/App.tsx [hunk 3]
```diff
@@ -27,7 +27,9 @@ export class AppBase extends React.Component<IAppProps, IAppState> {
     const { customizations } = appDefinition;
     const { isMenuVisible } = this.state;
 
-    const classNames = (this._classNames = getClassNames(styles, { responsiveMode, theme }));
+    const showOnlyExamples = location && location.hash.indexOf('docsExample=true') > -1;
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/App/App.tsx]: To account for SSR you have to check for `location` using `typeof window !== 'undefined' && window.location` (or preferably use the `getWindow()` utility).

**Comment by ecraig12345** [comment]
@Vitalius1 I don't think the router has been used in any of the SSR tests up to this point. I added a comment in the code with a suggestion about what to do.

File: packages/example-app-base/src/components/ComponentPage/ComponentPage.tsx (hunk 1 above)
Comment (Vitalius1) [reply packages/example-app-base/src/components/ComponentPage/ComponentPage.tsx]: I will clean up a code as soon as I have a working prototype to test in docs.

Comment (ecraig12345) [reply packages/example-app-base/src/components/ComponentPage/ComponentPage.tsx]: Fair enough!

File: packages/example-app-base/src/components/ComponentPage/ComponentPage.tsx (hunk 2 above)
Comment (Vitalius1) [reply packages/example-app-base/src/components/ComponentPage/ComponentPage.tsx]: I am planning to automatically generate a markdown file with an iframe pointing to this url along with the query parameter. So we are controlling the generation of those iframes and the `src` attribute. That is why I need a dev link to test it. I will try your suggestion to try and pass the ssr-tests and see how it goes.

File: packages/example-app-base/src/components/App/App.tsx (hunk 3 above)
Comment (Vitalius1) [reply packages/example-app-base/src/components/App/App.tsx]: 👍 will try and do that.

**Comment by dzearing** [comment]
@Vitalius1 can we close this or do you still need?

**Comment by Vitalius1** [comment]
We can close it for now and I will reopen one in the new master.

**Pull request closed without merging**